lib/logstash/inputs/elasticsearch.rb

```diff
@@ -668,6 +679,31 @@ def test_connection!
     raise LogStash::ConfigurationError, "Could not connect to a compatible version of Elasticsearch"
   end
 
+  def test_serverless_connection!
+    begin
+      @client.info(:headers => DEFAULT_EAV_HEADER ) if serverless?
+    rescue => e
+      @logger.error("Failed to retrieve Elasticsearch info", message: e.message, exception: e.class, backtrace: e.backtrace)
+      raise LogStash::ConfigurationError, "Could not connect to a compatible version of Elasticsearch"
+    end
+  end
```

Comment:
in these cases we can get rid of the level of indentation and just do:

```suggestion
  def test_serverless_connection!
    @client.info(:headers => DEFAULT_EAV_HEADER ) if serverless?
  rescue => e
    @logger.error("Failed to retrieve Elasticsearch info", message: e.message, exception: e.class, backtrace: e.backtrace)
    raise LogStash::ConfigurationError, "Could not connect to a compatible version of Elasticsearch"
  end
```